aptos-move/aptos-vm/src/sharded_block_executor/aggr_overridden_state_view.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use aptos_state_view::{StateView, TStateView};
+use aptos_types::{
+    state_store::{
+        state_key::StateKey, state_storage_usage::StateStorageUsage, state_value::StateValue,
+    },
+    write_set::TOTAL_SUPPLY_STATE_KEY,
+};
+
+pub const TOTAL_SUPPLY_AGGR_BASE_VAL: u128 = u128::MAX >> 1;
+#[derive(Clone)]
+pub struct AggregatorOverriddenStateView<'a, S> {
+    base_view: &'a S,
+}
+
+impl<'a, S: StateView + Sync + Send> AggregatorOverriddenStateView<'a, S> {
+    pub fn create_cross_shard_state_view_aggr_override(base_view: &'a S) -> Self {
```

Comment:
nit: `create_cross_shard_state_view_aggr_override` -> `new`

Reply:
Also, its probably cleaner to pass the `total_supply` in the constructor and store it in the struct.

Reply:
done